### ISSUE TITLE
idea: freeze command

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#4559]
Which given a subdataset would annotate it as frozen - no commits should be allowed to go in (at least via datalad), datalad status would skip introspecting it. We're might even just make its worktree read only explicitly (might want to become an option for install/clone).
Use case - analysis datasets which use heavy subdatasets (hcp, openneuro) as input. ATM any operation would be taking long time even if analysis dataset itself is actually quite lightweight. Caching would partially address it, but explicit read only mode would make it much faster.

**Comment by mih** [comment]
I think in the form you described such a feature is dangerous. The principle setup (read-only) datasets is typically used by my students, but unintentional modifications of such datasets are not uncommon. I rely on them being capture, in order to discover and rectify them. However, if datalad stops inspecting subdatasets merely because they are *declared* read-only, instead of actually being read-only, any modification would go unnoticed forever, and analysis results could be based on untracked modifications.

ATM git-annex does nothing to prevent modification of the worktree. With `status -e commit` subdataset introspection is fast. I do not trust the thought that (scientific) software can universally deal with a scenario where to-be-processed data lives in read-only directories -- but that may be overly pessimistic.

**Comment by yarikoptic** [comment]
Humans tend to shoot themselves in the foot over and over again and we can't stop them!  but we can facilitate some use cases to become more efficient. So if we mark some subdataset read-only and make even do make it read-only, I still think it could be quite of benefit

**Comment by mih** [comment]
True. I think that would be beneficial. I am not confident that we would need a command for it though. This would be for subdatasets only. What about a configuring superdatset (in .gitmodules) to make subdatasets read-only on clone? We already have a slot for post-clone processing. I don't think there is much benefit of freezing and unfreezing the same clone, but it would complicate implementation quite a bit.
